### PR TITLE
Update kpng-local-up.sh and fix potential IPVS panic

### PR DESCRIPTION
**Pull request by aryan9600**
While trying to run the project locally using `hack/kpng-local-up.sh`, I encountered several problems:
* Using the ipvs backend used to cause a panic, because the base distro being used in the kind node was out of date hence there was no `proc/sys/net/ipv4/vs` present, resulting in `Setup()` to fail silently. Later when a new service gets synced, `AddIP()` panics because `s.dummy` never got set.
* The `hack/kpng-local-up.sh` is out of date using `envsubst` instead of `j2`

### CODE REVIEW

File: hack/utils.sh [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    [ -d "${directory}" ]
+    if_error_exit "Directory \"${directory}\" does not exist"
+
+    case ":${PATH:-}:" in
+        *:${directory}:*) ;;
+        *) PATH="${directory}${PATH:+:$PATH}" ;;
+    esac
+}
+
+command_exists() {
```
Comment (astoycos) [inline hack/utils.sh:75]: Can you keep the comment here :)

Comment (aryan9600) [reply hack/utils.sh:75]: sorry it got lost somehow while refactoring, i added it back :)

File: hack/kpng-local-up.sh [hunk 2]
```diff
@@ -72,8 +70,18 @@ function install_k8s {
...
-    envsubst <kpng-deployment-ds.yaml.tmpl >kpng-deployment-ds.yaml
+    # Setting vars for generate the kpng deployment based on template
+    kpng_image="${IMAGE}" \
+    image_pull_policy="${PULL}" \
+    backend="${BACKEND}" \
+    config_map_name="${CONFIG_MAP_NAME}" \
+    service_account_name="${SERVICE_ACCOUNT_NAME}" \
+    namespace="${NAMESPACE}" \
+    e2e_backend_args="${BACKEND_ARGS}"\
+    j2 ./kpng-deployment-ds.yaml.j2 -o ./kpng-deployment-ds.yaml
```
Comment (astoycos) [inline hack/kpng-local-up.sh:83]: So I think @qabpea Just removed this? (I had added it originally) I'd like him to take a look as well

Comment (astoycos) [reply hack/kpng-local-up.sh:83]: I'm alright with just using j2 for now though!

**Comment by astoycos** [comment]
Thanks for doing this! I just want to see what Per has to say about re-using J2 since he just merged https://github.com/kubernetes-sigs/kpng/pull/345

**Comment by astoycos** [comment]
@aryan9600 Overall this looks great and I tested locally, do you mind also adding a quick note to the readme in `/hack/` Explaining how to spin up a devel cluster with `./hack/test_e2e.sh  -d -i ipv4 -b <backend>` and the difference between that and `./hack/kpng-local-up.sh`? 

I'll go ahead and give it an lgtm ig you want to tackle that in a follow up PR

/lgtm

File: hack/kpng-local-up.sh (hunk 2 above)
Comment (qabpea) [reply hack/kpng-local-up.sh:83]: Is the question where bgp2go is installed?
I did this
    echo "'bpf2go' not found, installing with 'go'"
	# set GOBIN to bin_directory to endure that binary is in search path
	export GOBIN=${install_directory}
	
	#remove GOPATH just to be sure
	export GOPATH=""


The problem I was solving was when a user do not have a GO environment setup, but go is installed and bgp2go is not in the path
go install  installs in the default GOBIN directory which is often not part of the users path.
GOBIN is therfor  set to the local bin dir.

But this should never be triggered unless there is no bgp2go in the path to begin with.

**Comment by astoycos** [comment]
Hrm 
```
# github.com/cilium/ebpf/internal
../../../go/pkg/mod/github.com/cilium/ebpf@v0.9.3-0.20220920164205-03b648ce0c4c/internal/deque.go:6:14: syntax error: unexpected any, expecting ]
../../../go/pkg/mod/github.com/cilium/ebpf@v0.9.3-0.20220920164205-03b648ce0c4c/internal/deque.go:12:16: syntax error: unexpected [, expecting comma or )
../../../go/pkg/mod/github.com/cilium/ebpf@v0.9.3-0.20220920164205-03b648ce0c4c/internal/deque.go:16:16: syntax error: unexpected [, expecting comma or )
../../../go/pkg/mod/github.com/cilium/ebpf@v0.9.3-0.20220920164205-03b648ce0c4c/internal/deque.go:21:16: syntax error: unexpected [, expecting comma or )
../../../go/pkg/mod/github.com/cilium/ebpf@v0.9.3-0.20220920164205-03b648ce0c4c/internal/deque.go:37:16: syntax error: unexpected [, expecting comma or )
../../../go/pkg/mod/github.com/cilium/ebpf@v0.9.3-0.20220920164205-03b648ce0c4c/internal/deque.go:40:2: syntax error: non-declaration statement outside function body
../../../go/pkg/mod/github.com/cilium/ebpf@v0.9.3-0.20220920164205-03b648ce0c4c/internal/deque.go:52:16: syntax error: unexpected [, expecting comma or )
../../../go/pkg/mod/github.com/cilium/ebpf@v0.9.3-0.20220920164205-03b648ce0c4c/internal/deque.go:53:6: zero redeclared in this block
	/home/runner/go/pkg/mod/github.com/cilium/ebpf@v0.9.3-0.20220920164205-03b648ce0c4c/internal/deque.go:38:6: previous declaration
../../../go/pkg/mod/github.com/cilium/ebpf@v0.9.3-0.20220920164205-03b648ce0c4c/internal/deque.go:55:2: syntax error: non-declaration statement outside function body
../../../go/pkg/mod/github.com/cilium/ebpf@v0.9.3-0.20220920164205-03b648ce0c4c/internal/deque.go:70:16: syntax error: unexpected [, expecting comma or )
../../../go/pkg/mod/github.com/cilium/ebpf@v0.9.3-0.20220920164205-03b648ce0c4c/internal/deque.go:53:6: too many errors
note: module requires Go 1.[18](https://github.com/kubernetes-sigs/kpng/actions/runs/3092361770/jobs/5003547330#step:3:19)
```

**Comment by astoycos** [comment]
/hold 

Just until  CI is green

**Comment by aryan9600** [comment]
@astoycos  I pinned `cillium/ebpf` to v0.9.2 in `hack/test_e2e.sh`, to avoid this, i'll do it in the github workflow as well

File: hack/kpng-local-up.sh (hunk 2 above)
Comment (astoycos) [reply hack/kpng-local-up.sh:83]: It looks like we're not changing that behavior here :)

**Comment by astoycos** [comment]
@jayunit100 This should be good :)

**Pull request merged**